packages/plugin-ext/src/common/rpc-protocol.ts

```diff
@@ -552,6 +552,10 @@ function safeStringify(obj: any, replacer?: JSONStringifyReplacer): string {
     try {
         return JSON.stringify(obj, replacer);
     } catch (err) {
+        console.warn('error stringifying response: ' + err);
+        if (err && err.stack) {
+            console.error(err.stack);
+        }
```

Comment:
Can we check for the error instead of checking if it has `stack`. Otherwise, you will log the stack trace twice, don't you?

```ts
} catch (err) {
    if (err instanceof SyntaxError) {
        console.error('Error stringifying response', err);
    } else {
        console.error(err);
    }
    return null;
}
```

Reply:
I don't really think we need to handle SyntaxError separately: the case I'm fixing here is on or a circular structure, not syntax.